src/Symfony/Component/Notifier/Bridge/Matrix/MatrixTransport.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Matrix;
+
+use Symfony\Component\Notifier\Bridge\Matrix\Exception\UnsupportedRecipientTypeException;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Component\Uid\Uuid;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Frank Schulze <frank@akiber.de>
+ */
+final class MatrixTransport extends AbstractTransport
+{
+    public function __construct(
+        #[\SensitiveParameter] private string $accessToken,
+        private bool $ssl,
+        ?HttpClientInterface $client = null,
+        ?EventDispatcherInterface $dispatcher = null,
+    ) {
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return \sprintf('matrix://%s', $this->getEndpoint(false));
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof MatrixOptions);
```

Comment:
```suggestion
        return $message instanceof ChatMessage;
```

for consistency with other bridges

Reply:
it is consistent with other bridges

https://github.com/symfony/symfony/blob/4dcb217c46b54633fcf2d2d75c2b552f363bc6d9/src/Symfony/Component/Notifier/Bridge/Discord/DiscordTransport.php#L50

https://github.com/symfony/symfony/blob/4dcb217c46b54633fcf2d2d75c2b552f363bc6d9/src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransport.php#L46

Reply:
We are not that consistent, sometimes we check for the options and sometimes not